homeassistant/components/tuya/climate.py

```diff
@@ -169,6 +169,10 @@ def __init__(
         # Figure out current temperature, use preferred unit or what is available
         celsius_type = self.find_dpcode(DPCode.TEMP_CURRENT, dptype=DPType.INTEGER)
         farhenheit_type = self.find_dpcode(DPCode.TEMP_CURRENT_F, dptype=DPType.INTEGER)
+        if celsius_type is None:
+            celsius_type = self.find_dpcode(DPCode.TEMP_UPPER, dptype=DPType.INTEGER)
+        if farhenheit_type is None:
+            celsius_type = self.find_dpcode(DPCode.TEMP_UPPER_F, dptype=DPType.INTEGER)
```

Comment:
I do feel like this is out of scope for this PR, as it adds new data points to solve a different issue?

Additionally, we should probably also use these for setting the temp limits on the climate entity as well in that case.

Reply:
this field contains the "current temperature" in my case:

`{"code":"upper_temp","lang_config":{"unit":"℃"},"name":"Current Temp","type":"Integer","values":"{\"unit\":\"℃\",\"min\":0,\"max\":100,\"scale\":0,\"step\":5}"}`

`{ "code": "upper_temp", "value": 34 }`

Reply:
According to documentation, it specifies the maximum temperature, not the current.

Anyways, this should not be in this PR IMHO, as it has no relation to the scaling calculations.

Reply:
As requested I removed those lines...
Although it might be neccassary submit a similar patch in the future

But technically the missing "current temperature" value should be fixed in a separate commit

Reply:
> According to documentation, it specifies the maximum temperature, not the current.
> 
> Anyways, this should not be in this PR IMHO, as it has no relation to the scaling calculations.

This is weird anyways because the min and max temperature are defined through min and max in "set_temp"
But anyway i removed those lines